Support three types of batched jdbc statements.

stoyle opened the pull request:
This is just a suggestion, I am fine with any changes you propose.

JDBC batching is kind of awkward imo, and it does not fit very good with kotliquery (either). So after trying a bit, I decided that very specific api calls may be best, i.e. not relying on Query at all. I think this will fit quite well with what we need in our project.

stoyle committed:
Support three types of batched jdbc statements.

1. Regular dump batched Statements
2. Prepared statements with positional params.
3. Prepares statements with kotliquery named params.

Currently doing this outside of kotliquery is a bit tedious
so would be nice to have this natively supported.

stoyle committed:
Allow nulls with Any? and prefer Collection over list.

stoyle committed:
Test positional params even further.

seratch commented on src/main/kotlin/kotliquery/Session.kt:
```diff
@@ -129,6 +154,26 @@ open class Session(
         return rows(query, extractor).toList()
     }
 
+    fun batchManyStatements(statements: Collection<String>): List<Int> {
```
OK, this method cannot accept prepared statements. Is there any way to describe those? What do you think about raw statements? I would like to have an explicitly different name from "statement" or "prepared statement" for it. If we go with raw statements here, the name of the method would be `fun batchRawStatements`.

stoyle replied:
This is not something we use, but is supported by jdbc. So I thought I'd add those, just for completeness. I have no problems with changing the name. Since this function does not take any query params, it implies it does not use prepared statements, but begin explicit is better.

We can of course just leave this out. I have never used such statements in real work work, and can be easily added if someone else would like it...

seratch commented on src/main/kotlin/kotliquery/Session.kt:
```diff
@@ -129,6 +154,26 @@ open class Session(
+    fun batchManyStatements(statements: Collection<String>): List<Int> {
+        warningForTransactionMode()
+        return using(connection.underlying.createStatement()) { st ->
+            statements.forEach{
```
nit: code formatting for `forEach{`

stoyle replied:
No problem, will fix :)

By the way, sorry for late reply. Gihub never notified my about you comments.

stoyle committed:
Better name for raw batched statements.

stoyle commented:
Rebased on changes from master, and updated according to comments. Still think we can leave out the `batchRawStatements` if you want, perhaps not a good code practice to use.

seratch commented:
@stoyle Thanks. LGTM. I agree to remove `batchRawStatements`. Let me delete the method after merging your PR. Thank you for your contribution! I will release a new version tonight, Japan Standard Time.

stoyle commented:
Awesome, thanks :)
